lib/neo4j/core/cypher_session/adaptors/faraday_helpers.rb

```diff
@@ -0,0 +1,49 @@
+module Neo4j
+  module Core
+    class CypherSession
+      module Adaptors
+        module FaradayHelpers
+          private
+
+          def verify_faraday_options(faraday_options = {}, defaults = {})
+            faraday_options.symbolize_keys!.reverse_merge!(defaults)
+            faraday_options[:adapter] ||= :net_http_persistent
+            require 'typhoeus/adapters/faraday' if faraday_options[:adapter].to_sym == :typhoeus
+            faraday_options
+          end
+
+          def set_faraday_middleware(faraday, options = {adapter: :net_http_persistent})
+            adapter = options.delete(:adapter)
+            send_all_faraday_options(faraday, options)
+            faraday.adapter adapter
+          end
+
+          def send_all_faraday_options(faraday, options)
+            options.each do |key, value|
+              next unless faraday.respond_to? key
+              if value.is_a? Array
+                if value.none? { |arg| arg.is_a? Array }
+                  faraday.send key, *value
+                else
+                  value.each do |args|
+                    arg_safe_send faraday, key, args
+                  end
+                end
+              else
+                faraday.send key, value
+              end
+            end
+          end
```

Comment:
I didn't look **too** hard, but I took [these comments](https://github.com/lostisland/faraday/blob/master/lib/faraday/connection.rb#L42-L56) to mean no since I don't see anything mentioning :response as a key, for instance.  But yeah, if that's true that makes life much easier - I'll give it a try